v2/contribute/decisions/session/0031-internal-prop-handling-in-flat-access-tokens.mdx

```diff
@@ -0,0 +1,62 @@
+---
+id: "0031"
+title: Do not add internal only claims to the front-token
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader";
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro";
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon";
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut";
+
+# Do not add internal only claims to the front-token
+
+<DecisionHeader
+  status="proposed"
+  lastUpdate="2023-01-22"
+  created="2023-01-22"
+  deciders={["rishabhpoddar", "porcellus"]}
+  proposedBy={["porcellus"]}
+/>
+
+## Context and Problem Statement
+
+When flattening the access token structure and making it into a standard JWT, we have to decide if it's safe to add all claims (including internally used ones) into the front-token, or not.
+
+## Considered Options
+
+- Hide internal-only claims
+- Hide all claims not added by the user
+- Expose all claims
+
+## Decision Outcome
+
+We should hide internal-only claims
+
+- We don't want to confuse our users
+- They are never used in applications
+
+We decided to hide: `refreshTokenHash1`, `parentRefreshTokenHash1`, because they do not make sense for the application and are never used as a part of CDI or FDI. 
```

Comment:
One issue with this is that it causes inconsistencies when reading the access token payload on the backend and frontend. The backend will show these "hidden" values, but the frontend wont.

Therefore, if there is no security risk in exposing these to the frontend (which i don't think there is), then we should just expose them.

Reply:
We could "hide" them from the `getAccessTokenPayload` on the backend; I assumed that both implementations would have to return the same thing.
Anyway, if you think these extra props are not too confusing, I'm happy with just exposing them all.

Reply:
Yea, i don't like the idea of hiding in general. Especially since the access token is a JWT and when someone sees it on a jwt inspector site, they will see those extra claims anyway